client/electron/go_helpers.ts

```diff
@@ -34,17 +34,19 @@ import {ChildProcessHelper} from './process';
  */
 export async function checkUDPConnectivity(
   config: string,
+  adapterIp: string = null,
+  adapterIndex: string = null,
   debugMode: boolean = false
 ): Promise<boolean> {
   const tun2socks = new ChildProcessHelper(pathToEmbeddedTun2socksBinary());
   tun2socks.isDebugModeEnabled = debugMode;
 
-  console.debug('[tun2socks] - checking connectivity ...');
-  const output = await tun2socks.launch([
-    '-transport',
-    config,
-    '-checkConnectivity',
-  ]);
+  const args = ['-transport', config, '-checkConnectivity'];
+  if (adapterIp && adapterIndex) {
+    args.push('-adapterIP', adapterIp, '-adapterIndex', adapterIndex);
```

Comment:
Why do you need to pass both the adapter IP and index? Can't you get one from the other?

Reply:
Removed IP.